src/swap/components/SwapProvider.tsx

```diff
@@ -70,6 +70,9 @@ export function SwapProvider({
     },
   }); // Component lifecycle
 
+  const [isToastVisible, setIsToastVisible] = useState(false);
+  const [transactionHash, setTransactionHash] = useState('');
```

Comment:
yeah i like that but we would do `setToastContent(transactionHash)` right ?

Reply:
hmm well we don't really need to store the whole component because theyre rendering `SwapToast` as a child (following the pattern of the rest of the components) right? also we could use the toast for other information by checking the `isToastVisible` value along with an error state, for example, stored in `lifeCycleStatus`. The reason i had to do the extra state for `transactionHash` is because we immediately reset the `lifeCycleStatus` to `init` after calling onsuccess. 

curious what the rest of the team thinks about this @0xAlec @cpcramer 

my thinking is that if we want to do something like you're suggesting we should follow the same pattern with TransactionToast and maybe that should be a follow up.

Reply:
I think it makes sense regardless to not have the toast showing content directly from lifecycle.  If we want to toast an error, we don't want the toast to clear if the error clears, but I don't want to end up with a bunch of separate states to hold different things we want to show in the toast.   my initial thought was that success toast could store transactionHash in its internal state and encapsulate the chainExplorer logic, and then just get displayed as a child of the toast.